### ISSUE TITLE
Entry number should not be suddenly increased a lot

### ISSUE THREAD

**Issue by rleungx** [tikv/pd#4481]
### What did you do?

When there is some reason which causes some leader transfer operations, the cache entry number increases so much.

![image](https://user-images.githubusercontent.com/35896542/146744098-c46f084d-aebf-413a-8558-d0e69e5d4d71.png)

<img width="922" alt="Screen Shot 2021-12-20 at 4 48 04 PM" src="https://user-images.githubusercontent.com/35896542/146743789-0e9190d8-050d-40f7-a6e0-3d9d23868687.png">

### What version of PD are you using (`pd-server -V`)?

master

**Comment by zhangjinpeng1987** [comment]
Are there any side effects?

**Comment by rleungx** [comment]
> Are there any side effects?

AFAIK, it won't have side effects or not too much. Since it will be back to normal when these regions receive the next heartbeat. /cc @lhy1024

**Comment by lhy1024** [comment]
> > Are there any side effects?
> 
> AFAIK, it won't have side effects or not too much. Since it will be back to normal when these regions receive the next heartbeat. /cc @lhy1024

yes, peer will be not likely to be scheduled by the scheduler until after it becomes a hotspot, not after it enters the cache.

It could be that a large number of balance leaders caused the data reported by tikv to be inaccurate, thus allowing them to be candidates. I will investigate it further.

**Comment by rleungx** [comment]
@lhy1024 Does it still exist?
